docs/MachineConfigDaemon.md

```diff
@@ -170,4 +170,66 @@ The updated list of optimized updates and behaviour (as of Openshift 4.8) is as
+
+## Config Drift Detection
+
+### Overview
+
+There may be situations when the on-disk state of a node may differ or "drift"
+from what is specified by the MachineConfig. This is usually (though not
+always) due to a cluster admin manually editing files. In the past, the MCD
+would only verify that the on-disk state matches what is specified by the
+MachineConfig after a reboot. Whenever this occurs, the node and Machine Config
+Pool (MCP) would be marked Degraded and unable to apply any MachineConfigs.
+
+Because reboots do not frequently occur, the config drift isn't immediately
+detected. In cases where the config drift was intentional, the cluster admin
+may lose the context which necessitated the config drift between the time the
+drift occurred and when it was detected. This leads to the node and / or MCP
+becoming degraded at an inconvenient time and can block future updates until
+the issue is remedied.
+
+### Detection
+
+To avoid this, as of 4.10, the MCD now has a Config Drift Monitor. It uses
+`fsnotify` to proactively detect config drift and mark the node `Degraded`
+within seconds of the drift taking place.
+
+Whenever a filesystem write event is detected for any of the objects (Ignition
+files and systemd units / dropins) defined in the currently applied
+MachineConfig, the Config Drift Monitor validates that the file contents and
+permissions fully match what the currently-applied MachineConfig specifies. In
+addition to this, the on-disk state will be verified immediately before a new
+MachineConfig is updated.
+
+Whenever the Config Drift Monitor detects an inconsistent object, it will:
+1. Emit an error to the console logs.
+1. Emit a Kubernetes event indicating that a configuration drift has occurred.
+1. Stop further verification.
+1. Set `machineconfiguration.openshift.io/state` to `Degraded`. 
+
+### Machine Config Updates
+
+To apply a new MachineConfig, the Config Drift Monitor is temporarily shut down.
+This is because the config will "drift" from the current MachineConfig to the
+new MachineConfig. Once new MachineConfig has been applied (assuming no reboot
+is needed), the Config Drift Monitor will be restarted and supplied with the
+newly active MachineConfig. Once the node has rebooted, the Config Drift Monitor
+is started once the MCD is out of the booting phase.
+
+Whenever the Config Drift Monitor is started or stoppped, a Kubernetes Event
+will be emitted. At startup, the event will also include the name of the
+MachineConfig it is using as a reference.
+
+### Remediation
+
+Once config drift is detected, there are two ways to recover:
+
+1. Ensure that the contents and file permissons of the file on disk match what
+the MachineConfig expects and ensure that the file permissions match what. This
```

Comment:
The sentence trails off after `match what.` , maybe just the first half of the sentence is enough?

Reply:
Good catch! I usually write better than this and should probably use a word processor in the future.